ui/src/app/store/machine/utils/storage.ts

```diff
@@ -317,7 +347,7 @@ export const isCacheSet = (disk: Disk | null): boolean =>
  * @param fs - the filesystem to check.
  * @returns whether the filesystem is a VMFS6 datastore
  */
-export const isDatastore = (fs: Filesystem | null): fs is Filesystem =>
+export const isDatastore = (fs: Filesystem | null): boolean =>
```

Comment:
Driveby fix. `fs is Filesystem` is incorrect - just because something isn't a datastore doesn't mean it's also not a filesystem.